### PR TITLE
chore(ci): restrict k8s workflow runs

### PR DESCRIPTION
**Pull request by neuronull**
Restricts the running of k8s tests.
We now run all versions once nightly on weeknights.
Otherwise,  run only the latest version.

To track the workflow, we will setup the slack GH App to subscribe to the workflow.

### CODE REVIEW

File: .github/workflows/k8s_e2e.yml [hunk 1]
```diff
@@ -18,10 +21,12 @@ on:
     types: [created]
   merge_group:
     types: [checks_requested]
+  schedule:
+    # At midnight UTC Tue-Sat
+    - cron: '0 0 * * 2-6'
```
Comment (jszwedko) [inline .github/workflows/k8s_e2e.yml:26]: How will we be notified if the scheduled build fails? I just want to make sure it doesn't slip through the cracks. Maybe we could take this opportunity to setup a Slack notification?

**Review by jszwedko — COMMENTED** [review]
One question about how we'll keep abreast of failures, otherwise 👍

File: .github/workflows/k8s_e2e.yml (hunk 1 above)
Comment (spencergilbert) [reply .github/workflows/k8s_e2e.yml:26]: 👀 Datadog Monitor 👀

Comment (neuronull) [reply .github/workflows/k8s_e2e.yml:26]: Doh, yeah that's kinda important lol.
I'll take a look at adding a Datadog Monitor.

Comment (jszwedko) [reply .github/workflows/k8s_e2e.yml:26]: In addition to looking at Datadog Monitors, I'd also evaluate https://github.blog/changelog/2022-12-06-github-actions-workflow-notifications-in-slack-and-microsoft-teams/ .

Comment (neuronull) [reply .github/workflows/k8s_e2e.yml:26]: After some research and discussion, we will try the GHA workflow notifications in slack first and pivot if needed.
There aren't any edits needed to the workflow file itself for that.

**Review by spencergilbert — APPROVED** [review]

**Pull request merged**